File: chap4/Echange.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let echange x y =
   let z = !x in
   x := !y;
   y := z

let _ =
  let a = ref 4 in
  let b = ref 7 in
  echange a b;
  print_int !a;
  print_string " ";
  print_int !b;
  print_newline ()

File: chap4/Echange.js
// Ces programmes sont sous licence CeCILL-B V1.
var a,b;
function echange (x, y) {
	var z; 
	z = x;
	x = y;
	y = z;
}
a = 4;
b = 7;
echange(a,b);
ISNprint(a);
ISNprint("   ");
ISNprintln(b);
